datadog/resource_datadog_security_monitoring_default_rule.go

```diff
@@ -0,0 +1,261 @@
+package datadog
+
+import (
+	"errors"
+	"fmt"
+	datadogV2 "github.com/DataDog/datadog-api-client-go/api/v2/datadog"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceDatadogSecurityMonitoringDefaultRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDatadogSecurityMonitoringDefaultRuleCreate,
+		Read:   resourceDatadogSecurityMonitoringDefaultRuleRead,
+		Update: resourceDatadogSecurityMonitoringDefaultRuleUpdate,
+		Delete: resourceDatadogSecurityMonitoringDefaultRuleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"rule_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of the rule.",
+				ForceNew:    true,
+			},
+
+			"case": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Cases of the rule, this is used to update notifications.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"status": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Status of the rule case to match.",
+						},
+						"notifications": {
+							Type:        schema.TypeList,
+							Required:    true,
+							Description: "Notification targets for each rule case.",
+							Elem:        &schema.Schema{Type: schema.TypeString},
+						},
+					},
+				},
+			},
+
+			"enabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Enable the rule.",
+			},
+
+			"disabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Disable the rule.",
+			},
+		},
+	}
+}
+
+func resourceDatadogSecurityMonitoringDefaultRuleCreate(d *schema.ResourceData, meta interface{}) error {
+	// create only updates an existing rule
+	err := resourceDatadogSecurityMonitoringDefaultRuleUpdate(d, meta)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(d.Get("rule_id").(string))
+
+	return nil
+}
```

Comment:
IIRC The way "fake" resource like this one are handled in official provider is to not define an `Update` method and have `ForceNew` attribute on all the schema elements.

Not sure about the benefits of doing this 🤷 Might help with the conflict @btoueg talked.